components/content_settings/core/browser/brave_content_settings_pref_provider.cc

```diff
@@ -57,9 +151,187 @@ bool BravePrefProvider::SetWebsiteSetting(
+  auto brave_cookies_iterator = PrefProvider::GetRuleIterator(
+      CONTENT_SETTINGS_TYPE_PLUGINS,
+      brave_shields::kCookies,
+      incognito);
+
+  auto old_rules = std::move(brave_cookie_rules_[incognito]);
+
+  while (brave_cookies_iterator && brave_cookies_iterator->HasNext()) {
+    auto rule = brave_cookies_iterator->Next();
+    bool shields_down_for_site = false;
+    if (IsActive(rule, shield_rules, &shields_down_for_site)) {
+      rules.push_back(CloneRule(rule, true));
+      brave_cookie_rules_[incognito].push_back(CloneRule(rule, true));
+    } else if (shields_down_for_site) {
+      rules.push_back(
+          Rule(ContentSettingsPattern::Wildcard(),
+               rule.primary_pattern,
+               ContentSettingToValue(CONTENT_SETTING_ALLOW)->Clone()));
+      brave_cookie_rules_[incognito].push_back(
+          Rule(ContentSettingsPattern::Wildcard(),
+               rule.primary_pattern,
+               ContentSettingToValue(CONTENT_SETTING_ALLOW)->Clone()));
+    }
+  }
+
+  // get the list of changes
+  std::vector<Rule> brave_cookie_updates;
+  for (const auto& new_rule : brave_cookie_rules_[incognito]) {
+    auto match = std::find_if(
+        old_rules.begin(),
+        old_rules.end(),
+        [&new_rule](const auto& old_rule) {
+          // we want an exact match here because any change to the rule
+          // is an update
+          return new_rule.primary_pattern == old_rule.primary_pattern &&
+                 new_rule.secondary_pattern == old_rule.secondary_pattern &&
+                 ValueToContentSetting(&new_rule.value) ==
+                    ValueToContentSetting(&old_rule.value);
+        });
+    if (match == old_rules.end()) {
+      brave_cookie_updates.push_back(CloneRule(new_rule));
+    }
+  }
+
+  // find any removed rules
+  for (const auto& old_rule : old_rules) {
+    auto match = std::find_if(
+        brave_cookie_rules_[incognito].begin(),
+        brave_cookie_rules_[incognito].end(),
+        [&old_rule](const auto& new_rule) {
+          // we only care about the patterns here because we're looking
+          // for deleted rules, not changed rules
+          return new_rule.primary_pattern == old_rule.primary_pattern &&
+                 new_rule.secondary_pattern == old_rule.secondary_pattern;
+        });
+    if (match == brave_cookie_rules_[incognito].end()) {
+      brave_cookie_updates.push_back(
+          Rule(old_rule.primary_pattern,
+               old_rule.secondary_pattern,
+               base::Value()));
+    }
+  }
+
+  // Notify brave cookie changes as CONTENT_SETTINGS_TYPE_COOKIES
+  if (content_type == CONTENT_SETTINGS_TYPE_PLUGINS) {
+    // PostTask here to avoid content settings autolock DCHECK
+    base::PostTaskWithTraits(
+          FROM_HERE,
+          {content::BrowserThread::UI, base::TaskPriority::USER_VISIBLE},
+          base::BindOnce(&BravePrefProvider::NotifyChanges,
+                         weak_factory_.GetWeakPtr(),
```

Comment:
maybe `base::ThreadTaskRunnerHandle::Get()->PostTask(FROM_HERE, base::BindOnce(&Task));` or something to avoid adding weak factory